hooks/useReferral.ts

```diff
@@ -0,0 +1,23 @@
+import { useState } from 'react'
+import { useSearchParams } from 'next/navigation'
+
+
+export function useReferral() {
+
+  const [referral, setReferral] = useState({})
+
+  const searchParams = useSearchParams();
+  const referralCode = searchParams ? searchParams.get('c') : undefined;
+  const referralAddress = searchParams ? searchParams.get('a') : undefined;
+  const referralSignature = searchParams ? searchParams.get('s') : undefined;
+  const hasReferral = referralCode && referralAddress && referralSignature;
+
+  if (hasReferral) {
+    setReferral({
+      'c': referralCode,
+      'a': referralAddress,
+      's': referralSignature
```

Comment:
maybe use 

```ts
    setReferral({
      code: referralCode,
      address: referralAddress,
      signature: referralSignature
   })
```

here?

Also you might need to define a type for the object passed to useState. Here's an idea:

```ts
type Referral = {
  code?: string
  address?: string
  signature?: string
}
```

And

```ts
  const [referral, setReferral] = useState<Referral>({})
```